src/main.cpp

```diff
@@ -863,9 +890,24 @@ PYBIND11_MODULE(_core, m)
 
   py::class_<workspace_with_logger_contexts>(m, "Workspace")
       .def(py::init(
-               [](const std::vector<std::string>& args, const std::optional<py::bytes>& bytes)
+               [](const std::vector<std::string>& args, const std::optional<py::bytes>& bytes,
+                   bool record_feature_names, bool record_metrics)
                {
                  auto opts = std::make_unique<VW::config::options_cli>(args);
+                 if (record_metrics)
+                 {
+                   // VW enables metrics by passing a file to write the metrics to.
+                   opts->insert("extra_metrics", "THIS_FILE_SHOULD_NOT_EXIST1");
```

Comment:
is the 1 supposed to be a `!`?

Reply:
Nah, a few lines below is `THIS_FILE_SHOULD_NOT_EXIST2`